komga/src/main/kotlin/org/gotson/komga/interfaces/rest/SeriesController.kt

```diff
@@ -332,6 +333,35 @@ class SeriesController(
       ?: throw ResponseStatusException(HttpStatus.NOT_FOUND)
   }
 
+  @PatchMapping(value = ["{seriesId}/thumbnail"], consumes = [MediaType.MULTIPART_FORM_DATA_VALUE], produces = [MediaType.IMAGE_JPEG_VALUE, MediaType.IMAGE_PNG_VALUE, MediaType.IMAGE_GIF_VALUE])
+  @ResponseStatus(HttpStatus.ACCEPTED)
+  fun patchCustomSeriesThumbnail(
+    @AuthenticationPrincipal principal: KomgaPrincipal,
+    @PathVariable(name = "seriesId") seriesId: String,
+    @RequestParam("image") image: MultipartFile
+  ) {
+    val series = seriesRepository.findByIdOrNull(seriesId) ?: throw ResponseStatusException(HttpStatus.NOT_FOUND)
+    try {
+      seriesLifecycle.addCustomThumbnailForSeries(series, image)
+    } catch (e: Exception) {
+      throw ResponseStatusException(HttpStatus.INTERNAL_SERVER_ERROR)
```

Comment:
Internal Server Errors are when you don't know what happened, but if you catch something, you know what happened, so you need more fine grained exception catching and error messages